src/controller/python/chip/server/ServerInit.cpp

```diff
@@ -0,0 +1,206 @@
+}
+#endif
+
+using PostAttributeChangeCallback = void (*)(EndpointId endpoint, ClusterId clusterId, AttributeId attributeId, uint8_t mask,
+                                             uint16_t manufacturerCode, uint8_t type, uint16_t size, uint8_t * value);
+
+class PythonServerDelegate // : public ServerDelegate
+{
+public:
+    void SetPostAttributeChangeCallback(PostAttributeChangeCallback cb)
+    {
+        // ChipLogProgress(NotSpecified, "callback %p", cb);
+        mPostAttributeChangeCallback = cb;
+    };
+    PostAttributeChangeCallback mPostAttributeChangeCallback = nullptr;
+};
+
+PythonServerDelegate gPythonServerDelegate;
+
+void pychip_server_set_callbacks(PostAttributeChangeCallback cb)
+{
+    // ChipLogProgress(NotSpecified, "setting cb");
+    gPythonServerDelegate.SetPostAttributeChangeCallback(cb);
+}
+
+void pychip_server_native_init()
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    int result;
+    int tmpErrno;
+
+    err = chip::Platform::MemoryInit();
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(DeviceLayer, "Failed to initialize CHIP stack: memory init failed: %s", chip::ErrorStr(err));
+    }
+
+    err = chip::DeviceLayer::PlatformMgr().InitChipStack();
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(DeviceLayer, "Failed to initialize CHIP stack: platform init failed: %s", chip::ErrorStr(err));
+    }
+
+    ConfigurationMgr().LogDeviceConfig();
+
+    PrintOnboardingCodes(chip::RendezvousInformationFlag(chip::RendezvousInformationFlag::kBLE));
+
+#if defined(PW_RPC_ENABLED)
+    chip::rpc::Init();
+    ChipLogProgress(NotSpecified, "PW_RPC initialized.");
+#endif // defined(PW_RPC_ENABLED)
+
+    chip::DeviceLayer::PlatformMgrImpl().AddEventHandler(EventHandler, 0);
+
+#if CONFIG_NETWORK_LAYER_BLE
+    chip::DeviceLayer::ConnectivityMgr().SetBLEDeviceName("RpiMatterDali"); // Use default device name (CHIP-XXXX)
+    chip::DeviceLayer::Internal::BLEMgrImpl().ConfigureBle(LinuxDeviceOptions::GetInstance().mBleDevice, false);
+    chip::DeviceLayer::ConnectivityMgr().SetBLEAdvertisingEnabled(true);
+#endif
+
+#if CHIP_DEVICE_CONFIG_ENABLE_WPA
+    if (LinuxDeviceOptions::GetInstance().mWiFi)
+    {
+        chip::DeviceLayer::ConnectivityMgrImpl().StartWiFiManagement();
+        if (!EnsureWifiIsStarted())
+        {
+            ChipLogError(NotSpecified, "Wi-Fi Management taking too long to start - device configuration will be reset.");
+        }
+    }
+#endif // CHIP_DEVICE_CONFIG_ENABLE_WPA
+
+    // parts from ChipLinuxAppMainLoop
+
+    uint16_t securePort   = CHIP_PORT;
+    uint16_t unsecurePort = CHIP_UDC_PORT;
+
+    // Init ZCL Data Model and CHIP App Server
+    chip::Server::GetInstance().Init(nullptr, securePort, unsecurePort);
+
+    // Initialize device attestation config
+    // SetDeviceAttestationCredentialsProvider(Examples::GetExampleDACProvider());
+
+    result   = pthread_create(&sPlatformMainThread, nullptr, PlatformMainLoop, nullptr);
```

Comment:
This is sort of reinventing `StartEventLoopTask`, but OK.

Reply:
@bzbarsky-apple this is as it is currently in https://github.com/project-chip/connectedhomeip/blob/53dd5833f08e5a6e4f19ed26442b5fb3510cf7ec/src/controller/python/chip/native/StackInit.cpp#L83. If it needs to be changed in this PR, then it should be changed in there as well.

Reply:
It doesn't need to be changed in this PR, but maybe file a followup issue to look into using `StartEventLoopTask` instead?

Reply:
https://github.com/project-chip/connectedhomeip/issues/11416